src/osd/OSD.cc

```diff
@@ -5834,8 +5834,40 @@ bool OSD::scrub_random_backoff()
   return false;
 }
 
+bool OSD::scrub_time_permit()
```

Comment:
If you had an optional arg that could over-ride the ceph_clock_now() call, the unites could test some realistic usage cases.